sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -277,7 +277,7 @@ Page<DataFeed> NextPageFunc(string nextLink, int? pageSizeHint)
         /// </returns>
         /// <exception cref="ArgumentNullException"><paramref name="dataFeed"/>, <paramref name="dataFeed"/>.Name, <paramref name="dataFeed"/>.DataSource, <paramref name="dataFeed"/>.Granularity, <paramref name="dataFeed"/>.Schema, <paramref name="dataFeed"/>.IngestionSettings, or <paramref name="dataFeed"/>.IngestionSettings.IngestionStartTime is null.</exception>
         /// <exception cref="ArgumentException"><paramref name="dataFeed"/>.Name is empty.</exception>
-        public virtual async Task<Response<string>> CreateDataFeedAsync(DataFeed dataFeed, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<DataFeed>> CreateDataFeedAsync(DataFeed dataFeed, CancellationToken cancellationToken = default)
```

Comment:
I guess /i am curious.. was this a specific request from users? or why the change?

Reply:
It may look like a change from our side, but in fact we're just aligning with what other languages are doing. According to Johan, we should call the extra GET and avoid exposing the ID we parsed from the location header if possible. Here's his full argument:

> Generally we follow location headers. I think I was confused in the review as to if the id was returned as a location header or if we were working around a scenario where we only got a partial resource back, or for an operation that returned a 204/No-content.  Following the location link (in the general case) gives you a chance to get the right etag etc.
In the general case, the location URL should be opaque to the client. If we have to parse it, it is not ideal. In practical terms, it would be very breaking if resources were created anywhere other than under the collection resource, so it is probably fairly safe for us to - in this specific case - parse the URL.

I personally still prefer the old approach, but I don't feel strongly about it.

Reply:
> should this be called CreateandGet or something that suggest that it also does a Get?

I prefer keeping it only as Create as this is the main purpose of the method. If the service team one day starts to return the full body of the created data feed in the request in addition to the location header, we could remove the extra `Get` request and we'd still have a valid name.

Reply:
Talked offline. Looks like there is another `beta` release so in the meantime the approach can be tested and presented to .NET architects to confirm the decision